app/Actions/ConfigureFrontendFramework.php

```diff
@@ -2,36 +2,77 @@
 
 namespace App\Actions;
 
-use App\Shell\Shell;
+use App\Shell;
+use Illuminate\Support\Facades\Config;
 
 class ConfigureFrontendFramework
 {
     use LamboAction;
 
     private $shell;
-
     private $laravelUi;
+    private $availableFrontends;
 
     public function __construct(Shell $shell, LaravelUi $laravelUi)
     {
         $this->shell = $shell;
         $this->laravelUi = $laravelUi;
     }
 
-    public function __invoke()
+    public function __invoke(array $availableFrontends)
     {
-        if (! config('lambo.store.frontend')) {
+        $this->availableFrontends = $availableFrontends;
+
+        if (! Config::get('lambo.store.frontend')) {
             return;
         }
 
-        $this->laravelUi->install();
+        app('console-writer')->logStep('Configuring frontend scaffolding');
+
+        if ($this->continueInstallation()) {
+            $this->laravelUi->install();
+
+            $process = $this->shell->execInProject(sprintf("php artisan ui %s%s", config('lambo.store.frontend'), $this->extraOptions()));
+
+            $this->abortIf(! $process->isSuccessful(), sprintf("Installation of %s UI scaffolding did not complete successfully.", config('lambo.store.frontend')), $process);
+
+            app('console-writer')->success(config('lambo.store.frontend') . ' ui scaffolding installed.');
+        }
+    }
+
+    private function continueInstallation(): bool
+    {
+        if ($this->validFrontend()) {
+            return true;
+        }
+
+        $configuredFrontend = $this->chooseFrontend();
```

Comment:
Can we possibly move this into a class that's responsible for validating and normalizing the user's input so that isn't having to happen inside this action class?